### DISCUSSION TITLE
fix: keyboard input (#83)

### DISCUSSION THREAD

**Discussion by sonht-0555** [thenick775/gbajs3#86] [Q&A] [answered]
Recently I saw that you have to downgrade to handle the problem of input (not allowing continuous deletion, input input is very slow) I solved it by [ remove keypress-keydown-keyup in _emscripten_set_keydown_callback_on_thread in mgba.js file ] I think there's no need to downgrade. 
<img width="1666" alt="Screenshot 2024-04-19 at 08 49 33" src="https://github.com/thenick775/gbajs3/assets/26568422/e90690f1-32bd-4e5d-bc4c-95efb738e114">

**Comment by thenick775** [comment, marked as answer]
That js file gets compiled based on my fork located [here](https://github.com/thenick775/mgba), which is compiled with `emcc`. So I don't manually modify the generated output file.

I do like the keyboard for desktop support, and usually it plays nice with me stopping keyboard events when modals are open:

See [here](https://github.com/thenick775/mgba/blob/feature/wasm/src/platform/wasm/main.c#L197) for how I do this today.

Upgrading to the latest `emcc` caused this not to work, which I believe is a bug.

Downgrading the emscripten compiler (emcc) solved the issue by fixing the disabling of events in my case. I assume it will be fixed eventually in the emscripten/core repo soon.

**Comment by thenick775** [comment]
What did you see in reference to slow inputs?

I've never experienced that.

I do know continuous deletion in form inputs works just fine.

**Reply by sonht-0555** [reply]
I was wrong, the input still works normally, only when deleting, it cannot be deleted continuously, and is interrupted. The problem is that the keydown events on mgba.js have an impact

**Reply by thenick775** [reply]
Do you have a video of what your seeing?

From what I can see, continuous deletion is functional:

https://github.com/thenick775/gbajs3/assets/40526638/acf9bffb-1db6-4dd6-b329-16e1520aab62
